code/processes/segmentedtickerplant.q

```diff
@@ -83,5 +83,38 @@ init:{
   .stplg.init[string .proc.procname];
  };
 
-// Have the init function called from torq.q
+// Loads the striping.json config file and logs any problems
+configload:{[scpath]
+     @[jsonchecks; 
+       .stpps.stripeconfig:@[{.j.k read1 x};scpath;{.lg.o[`configload;"Failed to load in striping.json file: ",x]}];
+       {.lg.o[`configload;"Datastriping is not enabled due to problems with striping.json"]}]
+     };
+
+// Checks if each subscriptiondefault is set for each segment and errors if not defined
+jsonchecks:{[.stpps.stripeconfig]
+     // check defaults are ignore or all
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
```

Comment:
I'm not 100% sure of the structure of `.stpps.stripeconfig` but can we do something like

```suggestion
     defaults:first each .stpps.stripeconfig[;`subscriptiondefault];
```

?

Maybe needs a flip in there somewhere or something, but I think it should be possible to at least avoid iterating over the keys

Reply:
This is the structure of the .stpps.stripeconfig
![stripeconfigstructure](https://user-images.githubusercontent.com/85632530/179246476-603d78cb-3fd1-4871-9733-3877cfbc5350.JPG)
This line will do the same thing as original lie 96:
value first each .stpps.stripeconfig[;`subscriptiondefault]